### PR TITLE
Support rectangle selection in viewer

### PR DESCRIPTION
**Pull request by chungmin99**
Related to rect-select viser PR, linked here: https://github.com/nerfstudio-project/viser/pull/157. 

This adds `ViewerRectSelect`, a rectangular selection in 2D parameterized by its min/max bounds. This PR also adds `screen_pos` property to `ViewerClick`, the screen position associated with the click.

There's many exciting applications for this -- especially in splatfacto, where we can select groups of gaussians. One example is #3001, for floater removal (@anniezhang2288 :) )

The example video shows both interaction types, clicking and rectangular selection. More specifically, for the rectangular selection, the video shows:
* how gaussians can be selected -- only keeping the gaussians that project inside the selection box.
* ... and what happens if you continue training.

https://github.com/nerfstudio-project/nerfstudio/assets/10284938/6bfdfb99-c318-4cfc-a272-ac76e4aa899f

### CODE REVIEW

**Comment by brentyi** [comment]
Question: does it make sense to move toward deprecating the `ViewerElement` API? Would it make sense to give models access to the `ViserServer` instance directly?

(I'm not super familiar with the viewer elements workflow)

File: nerfstudio/viewer/viewer_elements.py [hunk 1]
```diff
@@ -60,17 +61,30 @@ class ViewerClick:
     The direction of the click if projected from the camera through the clicked pixel,
     in world coordinates
     """
+    screen_pos: Tuple[float, float]
+    """The screen position of the click in screen coordinates."""
```
Comment (kerrj) [inline nerfstudio/viewer/viewer_elements.py:65]: Maybe specify that this is also normalized [0,1]?

**Comment by kerrj** [comment]
> Question: does it make sense to move toward deprecating the `ViewerElement` API? Would it make sense to give models access to the `ViserServer` instance directly?
> 
> (I'm not super familiar with the viewer elements workflow)

Technically you can already access the viser API through `ViewerControl.viser_server`, although it's definitely messy since then theres a divide between the pre-defined buttons and the dynamically added buttons.

Making the viewer_elements abstraction thinner somehow would be nice, but we do need some sort of abstraction if we want the viewer to have the nice auto-populating behavior that it does right now (recursively scrape the pipeline for viewer elements).

The main annoyance with accessing the ViserServer is how to treat the "static" (predefined in model file) gui elements vs the "dynamic" (added during runtime) elements. Which panel do they go on, what order, etc. And how do we keep the same API for both?

File: nerfstudio/viewer/viewer_elements.py (hunk 1 above)
Comment (chungmin99) [reply nerfstudio/viewer/viewer_elements.py]: Resolved in https://github.com/nerfstudio-project/nerfstudio/pull/3014/commits/d9e6c889ad7efde630ec8497c1818b3168cf6a4e !

File: nerfstudio/viewer/viewer_elements.py [hunk 2]
```diff
@@ -147,42 +161,72 @@ def get_camera(self, img_height: int, img_width: int, client_id: Optional[int] =
         return get_camera(camera_state, img_height, img_width)
 
     def register_click_cb(self, cb: Callable):
+        """Deprecated, use register_pointer_cb instead."""
+        CONSOLE.log("`register_click_cb` is deprecated, use `register_pointer_cb` instead.")
+        self.register_pointer_cb("click", cb)
+
+    def register_pointer_cb(
```
[the remark this answers is not in the document]
Comment (chungmin99) [reply nerfstudio/viewer/viewer_elements.py:168]: Viser should raise warnings if a callback is unregistered, see commit here: https://github.com/nerfstudio-project/viser/pull/157/commits/2beac5937838aa1d8575d2319c9d61839e66fedf

... do you think we should throw an error? 

Example warning:
image

[the remark this answers is not in the document]
Comment (kerrj) [reply nerfstudio/viewer/viewer_elements.py:168]: I think it'd be a bit cleaner if there was a warning from ViewerControl too for traceability yea, if that's easy

[the remark this answers is not in the document]
Comment (chungmin99) [reply nerfstudio/viewer/viewer_elements.py:169]: Ok, addressed in [ec21904](https://github.com/nerfstudio-project/nerfstudio/pull/3014/commits/ec21904b24e64b9dfea2d946ea14d3efc97258c6) via catching warnings; now it prints like the img below:

image

So it should at least fwd you to the correct nerfstudio function.

**Review by kerrj — APPROVED** [review]
exciting stuff, lgtm :)

**Pull request merged**